azext_iot/dps/providers/device_registration.py

```diff
@@ -0,0 +1,268 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from logging import getLogger
+from typing import TypeVar
+from azext_iot.common.shared import AttestationType
+from azext_iot.constants import IOTDPS_PROVISIONING_HOST
+from azext_iot.dps.common import (
+    DISABLED_REGISTRATION_ERROR,
+    FAILED_REGISTRATION_ERROR,
+    UNAUTHORIZED_ERROR,
+    COMPUTE_KEY_ERROR,
+    CERTIFICATE_FILE_ERROR,
+    CERTIFICATE_RETRIEVAL_ERROR,
+    TPM_SUPPORT_ERROR,
+)
+from azext_iot.dps.providers.discovery import DPSDiscovery
+from azext_iot.operations.dps import (
+    iot_dps_compute_device_key,
+    iot_dps_device_enrollment_get,
+    iot_dps_device_enrollment_group_get
+)
+from azure.cli.core.azclierror import (
+    InvalidArgumentValueError,
+    RequiredArgumentMissingError,
+    AzureResponseError,
+    AzureConnectionError,
+    UnauthorizedError
+)
+
+logger = getLogger(__name__)
```

Comment:
We usually use knack.log.get_logger() for setting the logger object in a module, does logging.getLogger() work as expected and cover the configuration as the knack logging setup?